src/Stryker.Core/Stryker.Core.UnitTest/Mutants/CsharpMutantOrchestratorTests.cs

```diff
@@ -1948,4 +1951,315 @@ public void ShouldIncrementMutantCountUniquely()
+                     // Initialize private field:
+                     // Stryker disable String : Not mutation under test
+                     private static readonly ImmutableArray<string> _months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+                     // property with expression body:
+                     public IEnumerable<int> MaxDays =>
+                         [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+                     public int Sum(IEnumerable<int> values) =>
+                         values.Sum();
+                     public void Example()
+                     {
+                         // As a parameter:
+                         int sum = Sum([1, 2, 3, 4, 5]);
+                     }
+                     """;
+
+        var expected =
+            """
+            // Initialize private field:
+            // Stryker disable String : Not mutation under test
+            private static readonly ImmutableArray<string> _months =
+                StrykerNamespace.MutantContext.TrackValue(
+                    () =>(StrykerNamespace.MutantControl.IsActive(0) ? [] : ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]));
+
+            // property with expression body:
+            public IEnumerable<int> MaxDays =>
+                (StrykerNamespace.MutantControl.IsActive(13)
+                     ? []
+                     : [ 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 ]);
+
+            public int Sum(IEnumerable<int> values) =>
+                (StrykerNamespace.MutantControl.IsActive(14) ? values.Max() : values.Sum());
+
+            public void Example() {
+              if (StrykerNamespace.MutantControl.IsActive(15)) {
+              } else {
+                // As a parameter:
+                int sum = Sum(
+                    (StrykerNamespace.MutantControl.IsActive(16) ? [] : [ 1, 2, 3, 4, 5 ]));
+              }
+            }
+            """;
+        ShouldMutateSourceInClassToExpected(source, expected);
+    }
+
+    [TestMethod]
+    public void ShouldMutateNonConstatCollectionExpression()
+    {
+        var source = """
+                     public void M() {
+                         // Stryker disable String : Not mutation under test
+                         string hydrogen = "H";
+                         string helium = "He";
+                         string lithium = "Li";
+                         string beryllium = "Be";
+                         string boron = "B";
+                         string carbon = "C";
+                         string nitrogen = "N";
+                         string oxygen = "O";
+                         string fluorine = "F";
+                         string neon = "Ne";
+                         string[] elements = [hydrogen, helium, lithium, beryllium, boron, carbon, nitrogen, oxygen, fluorine, neon];
+                         foreach (var element in elements)
```

Comment:
I think we should strive to keep these tests as simple and clear as possible. In this case we can remove the foreach to only test the collection expression